src/gui_common/charts/line/LineChart.cs

```diff
@@ -577,8 +577,12 @@ private void CreateDropDownLegend(string title)
     /// </summary>
     private void RenderChart()
     {
-        if (dataSets.Count <= 0)
+        // Handle empty or entirely hidden datasets
+        if (dataSets?.All(d => !d.Value.Draw) == true)
```

Comment:
Good catch, just tested and it wasn't working anymore for null cases.
```suggestion
        if (dataSets?.All(d => !d.Value.Draw) != false)
```

Reply:
Fixed it.